### PR TITLE
Adds ability to extract metadata from Google BigQuery using SQLAlchemy

### PR DESCRIPTION
**Pull request by avelasquezhe84**
feat – Adds ability to extract metadata from Google BigQuery using SQLAlchemy

### Summary of Changes

Based on the Metadata Extractor for Snowflake, I created a metadata extractor for Google BigQuery, which queries the INFORMATION_SCHEMA instead of looping through all the projects and tables in the database.

1. Addition of the CREDS_PATH parameter to the databuilder/databuilder/extractor/sql_alchemy_extractor.py file
2. Addition of required libraries to the databuilder/setup.py file
3. Creation of the databuilder/databuilder/extractor/bigquery_sql_alchemy_metadata_extractor.py file
4. Creation of the databuilder/example/scripts/sample_bigquery_sql_alchemy_data_loader.py implementation file.

### Tests

Implemented this feature successfully in a client's project.

### Documentation

Similar to the Snowflake Metadata Extractor, in order to extract metadata from BigQuery using this method, the following parameters need to be provided by modifying the databuilder/example/scripts/sample_bigquery_sql_alchemy_data_loader.py file:

- BIGQUERY_PROJECT_KEY = 'YourBigqueryProjectId'
- BIGQUERY_TABLE_SCHEMA_KEY = 'YourBigqueryTableSchemaName'
- BIGQUERY_CONNECTION_STRING = f'bigquery://{BIGQUERY_PROJECT_KEY}'
- BIGQUERY_CREDENTIALS_PATH = ''

Refer this doc: https://github.com/googleapis/python-bigquery-sqlalchemy for supported connection parameters and configurations

### CheckList
- [X] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [X] PR includes a summary of changes.
- [X] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [X] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by feng-tao** [comment]
cc @xuan616

File: databuilder/databuilder/extractor/bigquery_sql_alchemy_metadata_extractor.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        self.sql_stmt = BigQueryMetadataExtractor.SQL_STATEMENT.format(
+            where_clause_suffix=conf.get_string(BigQueryMetadataExtractor.WHERE_CLAUSE_SUFFIX_KEY),
+            project=self._bigquery_project,
+            table_schema=self._bigquery_table_schema,
+            schema=self._bigquery_schema
+        )
+
+        LOGGER.info('SQL for bigquery metadata: %s', self.sql_stmt)
+
+        self._alchemy_extractor = sql_alchemy_extractor.from_surrounding_config(conf, self.sql_stmt)
```
Comment (xuan616) [inline databuilder/databuilder/extractor/bigquery_sql_alchemy_metadata_extractor.py:100]: does it support only keyfile path for now

**Comment by avelasquezhe84** [comment]
That's correct

**Comment by xuan616** [comment]
LGTM. Hi @avelasquezhe84 Can I know what is the difference from the current [bigquery_metadata_extrator](https://github.com/amundsen-io/amundsen/blob/main/databuilder/databuilder/extractor/bigquery_metadata_extractor.py) in terms of the results extracted when we're using sqlalchemy/information_schema, like the filter for sharded tables. Thanks

BTW, it would be better if there are unit tests for this new extractor. :smiley:

**Comment by feng-tao** [comment]
+1 on the unit test

File: databuilder/databuilder/extractor/bigquery_sql_alchemy_metadata_extractor.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            lower(c.DATA_TYPE) AS col_type,
+            ifnull(c.ORDINAL_POSITION, 0) AS col_sort_order,
+            'bigquery' AS database,
+            lower(c.TABLE_CATALOG) AS cluster,
+            lower(c.TABLE_SCHEMA) AS schema,
+            lower(c.TABLE_NAME) AS name,
+            tops.OPTION_VALUE AS description,
+            CASE c.IS_PARTITIONING_COLUMN WHEN 'YES' THEN 1 WHEN 'NO' THEN 0 ELSE NULL END AS is_partition_col,
+            CASE WHEN t.TABLE_TYPE = 'VIEW' THEN 0 ELSE 1 END AS is_view
+        FROM `{project}.{table_schema}.{schema}`.COLUMNS c
```
Comment (apalmercari) [inline databuilder/databuilder/extractor/bigquery_sql_alchemy_metadata_extractor.py:45]: How to handle different zones?

**Pull request closed without merging**